compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -999,11 +999,15 @@ impl<'a> FunctionContext<'a> {
         mut incremented_params: Vec<(ValueId, ValueId)>,
         terminator_args: &[ValueId],
     ) {
+        // TODO: This check likely leads to unsoundness.
```

Comment:
Should we link to our existing issue about reworking RCs?

Reply:
No, I think this unsoundness issue is separate. I made an issue here: https://github.com/noir-lang/noir/issues/7737. I should have a fix for it today.